mittab/apps/tab/archive.py

```diff
@@ -0,0 +1,148 @@
+            })
+
+            if debate.victor == Round.GOV_VIA_FORFEIT or \
+                    debate.victor == Round.ALL_DROP:
+                opp_tag.set("forfeit", "True")
+            elif debate.victor == Round.OPP_VIA_FORFEIT or \
+                    debate.victor == Round.ALL_DROP:
+                gov_tag.set("forfeit", "True")
+
+            team_points = [0, 0] # Gov, Opp
+            stats = sorted(list(debate.roundstats_set.all().values()),
+                           key=lambda x: pos.index(x["debater_role"]))
+            for i, speech in enumerate(stats):
+                side_tag = gov_tag if i % 2 == 0 else opp_tag
+                team_points[i % 2] += speech["speaks"]
+                speech_tag = SubElement(side_tag, "speech", {
+                    "speaker": SPEAKER_ID_PREFIX + str(speech["debater_id"]),
+                    "reply": "False"
+                })
+                ballot = SubElement(speech_tag, "ballot", {
+                    "adjudicators": adjs,
+                    "rank": str(speech["ranks"]),
+                    "ignored": "False"
+                })
+                ballot.text = str(speech["speaks"])
+
+            gov_win = [Round.GOV, Round.GOV_VIA_FORFEIT, Round.ALL_WIN]
+            gov_rank = "1" if debate.victor in gov_win else "2"
+            gov_ballot = SubElement(gov_tag, "ballot", {
+                "adjudicators": adjs, "rank": gov_rank})
+            gov_ballot.text = str(team_points[0])
+
+            opp_win = [Round.OPP, Round.OPP_VIA_FORFEIT, Round.ALL_WIN]
+            opp_rank = "1" if debate.victor in opp_win else "2"
+            opp_ballot = SubElement(opp_tag, "ballot", {
+                "adjudicators": adjs, "rank": opp_rank})
+            opp_ballot.text = str(team_points[1])
+
+    def add_participants(self):
+        participants_tag = SubElement(self.root, "participants")
+
+        for team in Team.objects.all().prefetch_related("debaters"):
+            team_tag = SubElement(participants_tag, "team", {
+                "id": TEAM_ID_PREFIX + str(team.id),
+                "name": team.name
+            })
+
+            institutions = SCHOOL_ID_PREFIX + str(team.school_id)
+            if team.hybrid_school_id is not None:
+                institutions += " " + SCHOOL_ID_PREFIX + str(team.hybrid_school_id)
+
+            for debater in team.debaters.all():
+                debater_tag = SubElement(team_tag, "speaker", {
+                    "id": SPEAKER_ID_PREFIX + str(debater.id),
+                    "categories": SPEAKER_STATUS_PREFIX + str(debater.novice_status),
+                    "institutions": institutions
+                })
+                debater_tag.text = debater.name
+
+        for judge in Judge.objects.all().prefetch_related("schools"):
+            SubElement(participants_tag, "adjudicator", {
+                "id": JUDGE_ID_PREFIX + str(judge.id),
+                "name": judge.name,
+                "score": str(judge.rank),
```

Comment:
So, on APDA, judge ranks aren't frequently re-calculated. They're usually just assigned once based on experience and a calibration ballot. I think most tab directors would be uncomfortable with the judge ranks they assign being released in any sort of public format. Would it be possible to leave this out?

Reply:
Or maybe I can just filter out this data before it's automatically released anywhere public?

Reply:
Thought the judge ranks would be useful for the tab directors to keep in their records, and that's why I added it. It's true however that public disclosure may not always be wanted. It would be easy to filter out after-generation as it is the only attribute called `score`.